File: 03-grokking-algorithms/02_selectionsort.py
def findSmallest(arr):
    smallest_index = 0
    smallest = arr[smallest_index]
    for i in range(1,len(arr)):
        if smallest > arr[i] :
            smallest = arr[i]
            smallest_index = i
    return smallest_index

def selectionSort(arr_original):
    arr_copy = list(arr_original)
    arr_sorted = []
    while len(arr_copy) > 0:
        smallest_index = findSmallest(arr_copy)
        smallest = arr_copy[smallest_index]
        arr_sorted.append(smallest)
        arr_copy.pop(smallest_index)       
    return arr_sorted


print(selectionSort([5, 3, 6, 2, 10]))

File: 03-grokking-algorithms/02_selection_sort.js
const findSmallest = (arr) => {
    let smallestIndex = 0
    let smallest = arr[smallestIndex]
    for (let i = 0; i < arr.length; i++){
        if (smallest > arr[i]) {
            smallest = arr[i]
            smallestIndex = i
        }
    }
    return smallestIndex
}

const selectionSort = (arr) => {
    const arr_copy = arr.concat()
    const arr_sorted = []
    while (arr_copy.length > 0){
        let smallestIndex = findSmallest(arr_copy)
        arr_sorted.push(arr_copy[smallestIndex])
        arr_copy.splice(smallestIndex, 1)        
    }
    return arr_sorted

}

console.log(selectionSort([5, 3, 6, 2, 10]));